lib/controllers/boxesController.js

```diff
@@ -79,14 +83,14 @@ const { config, Honeybadger } = utils,
  */
 const updateBox = function updateBox (req, res, next) {
   const deleteMeasurementsOf = [];
-  const { mqtt, ttn, loc, image, sensors, boxId, addons } = req._userParams;
-  let shouldSendSketch = false;
+  const { mqtt, ttn, location, image, sensors, boxId, addons } = req._userParams;
+  let shouldSendSketch = false, validLocation = null;
 
   if (sensors && addons) {
     return next(new restify.BadRequestError('sensors and addons can not appear in the same request.'));
   }
 
-  Box.findById(boxId).then(function (box) {
+  Box.findById(boxId, Box.BOX_PROPS_FOR_POPULATION).then(function (box) {
```

Comment:
I don't know if you're willing to do this, but I would prefer if the last location was persisted as plain values in the box not as an id. For example for calling /boxes currently, this would save 513 roundtrips to the database.

I understand that currently the lastMeasurement of the sensors cause many more roundtrips right now, but I plan to change the schema to store the lastMeasurements directly in the box documents.

Maybe there is a good reason for you to do it like this right now..